### PR TITLE
Updated to not use the reverse method

### PR DESCRIPTION
**Pull request by rayliaw**

### CODE REVIEW

File: RaymondLiaw-rayliaw/reverse_it.rb [hunk 1]
```diff
@@ -1,17 +1,33 @@
-# Checks to see if 'str' is a palindrome
+# Checks to see if parameter 'str' is a palindrome
 def is_palindrome(str)
-  # Remove spaces and saves a reversed copy. Downcase string for comparison
-  original = str.delete(' ').downcase
-  flipped = original.reverse
+  # Downcases, strips out white space and punctuation, and converts to an array of chars
+  original = str.downcase.gsub(/[^a-z0-9]/i, '').chars
```
Comment (tgaff) [inline RaymondLiaw-rayliaw/reverse_it.rb]: nice way to clear whitespace so yours can handle multiple words

File: RaymondLiaw-rayliaw/reverse_it.rb [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+# Returns a new array in the reversed order without using the reverse method
+def reverse_array(arr)
+  i = 0
+  flipped_arr = Array.new(arr.length)
+
+  # Loop through the passed array and save to a new array in reverse order
+  while i < arr.length
+    flipped_arr[arr.length-1-i] = arr[i]
+    i += 1
+  end
```
Comment (tgaff) [inline RaymondLiaw-rayliaw/reverse_it.rb]: Have you worked with another programming language before?  A more typical ruby way to do this is `arr.each { |char| .....`.  We'll likely talk more about that soon.

**Comment by tgaff** [comment]
Great job Ray!  :sunny:

File: RaymondLiaw-rayliaw/reverse_it.rb (hunk 2 above)
Comment (rayliaw) [inline RaymondLiaw-rayliaw/reverse_it.rb]: Would this work well using .each_index to traverse both arrays?
```ruby
arr.each_index { |e| flipped_arr[arr.length-1-e] = arr[e] }
```

Or is there a good way to use .each and not create another index for the 2nd array? As opposed to this..
```ruby
i = 0
...
arr.each do |e|
    flipped_arr[arr.length-1-i] = e
    i += 1
end
```

**Pull request merged**